pkg/operator/replacement_starter.go

```diff
@@ -0,0 +1,323 @@
+}
+
+func (a authenticationOperatorStarter) RunOnce(ctx context.Context) error {
+	if err := a.startInformers(ctx); err != nil {
+		return err
+	}
+
+	errs := []error{}
+
+	for _, controllerRunOnceFn := range a.controllerRunOnceFns {
+		// TODO add timeout.
+		syncCtx := factory.NewSyncContext("run-once-sync-context", a.eventRecorder)
+		err := controllerRunOnceFn(ctx, syncCtx)
+		errs = append(errs, err)
+	}
+	return errors.Join(errs...)
+}
+
+func (a authenticationOperatorStarter) startInformers(ctx context.Context) error {
+	for _, informer := range a.informers {
+		informer.Start(ctx)
+	}
+	for _, informer := range a.informers {
+		informer.WaitForCacheSync(ctx)
+	}
+	return nil
+}
+
+func (a authenticationOperatorStarter) Start(ctx context.Context) error {
+	if err := a.startInformers(ctx); err != nil {
+		return err
+	}
+	for _, controllerRunFn := range a.controllerRunFns {
+		go controllerRunFn(ctx, 1)
+	}
+	return nil
+}
+
+var _ OperatorStarter = &authenticationOperatorStarter{}
+
+const componentName = "cluster-authentication-operator"
+
+func CreateOperatorInputFromMOM(ctx context.Context, momInput libraryapplyconfiguration.ApplyConfigurationInput) (*AuthenticationOperatorInput, error) {
+	kubeClient, err := kubernetes.NewForConfigAndClient(&rest.Config{}, momInput.MutationTrackingClient.GetHTTPClient())
+	if err != nil {
+		return nil, err
+	}
+
+	configClient, err := configclient.NewForConfigAndClient(&rest.Config{}, momInput.MutationTrackingClient.GetHTTPClient())
+	if err != nil {
+		return nil, err
+	}
+
+	operatorClient, err := operatorclient.NewForConfigAndClient(&rest.Config{}, momInput.MutationTrackingClient.GetHTTPClient())
+	if err != nil {
+		return nil, err
+	}
+
+	authenticationOperatorClient, dynamicInformers, err := genericoperatorclient.NewOperatorClientWithClient(
+		momInput.Clock,
+		momInput.MutationTrackingClient.GetHTTPClient(),
+		operatorv1.GroupVersion.WithResource("authentications"),
+		operatorv1.GroupVersion.WithKind("Authentication"),
+		extractOperatorSpec,
+		extractOperatorStatus,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	eventRecorder := events.NewKubeRecorderWithOptions(
+		kubeClient.CoreV1().Events("openshift-authentication-operator"),
+		events.RecommendedClusterSingletonCorrelatorOptions(),
+		componentName,
+		&corev1.ObjectReference{
+			Kind:      "Deployment",
+			Namespace: "openshift-authentication-operator",
+			Name:      "authentication-operator",
+		},
+	)
+
+	return &AuthenticationOperatorInput{
+		kubeClient:                   kubeClient,
+		configClient:                 configClient,
+		operatorClient:               operatorClient,
+		authenticationOperatorClient: authenticationOperatorClient,
+		eventRecorder:                eventRecorder,
+		informers: []SimplifiedInformerFactory{
+			DynamicInformerFactoryAdapter(dynamicInformers), // we don't share the dynamic informers, but we only want to start when requested
```

Comment:
why not to move `DynamicInformerFactoryAdapter` to a common place ? I think it will be needed for all operators, right?